pki/pki.go

```diff
@@ -56,15 +57,39 @@ func (p *PKI) Config() any {
 	return &p.config
 }
 
-func (p *PKI) Configure(_ core.ServerConfig) error {
+func (p *PKI) Configure(config core.ServerConfig) error {
 	var err error
 	p.validator, err = newValidator(p.config)
 	if err != nil {
 		return err
 	}
+	trustStore, err := loadTrustStore(config.TLS.TrustStoreFile)
+	if err != nil {
+		return err
+	}
+	if trustStore != nil {
+		err = p.addCAs(trustStore.Certificates())
```

Comment:
true, but this will only be the case for the first leaf cert presented for an unknown chain. So it just adds a bit of data one time that cannot do any harm. It does allow prefetching of its issuers CRL since that can only be found on the leaf cert.